go/vt/vttablet/tabletmanager/healthcheck.go

```diff
@@ -286,17 +287,15 @@ func (agent *ActionAgent) runHealthCheckLocked() {
 	agent.History.Add(record)
 
 	// try to figure out the mysql port if we don't have it yet
-	if topoproto.MysqlPort(tablet) == 0 && !agent.skipMysqlPortCheck {
-		// we don't know the port, try to get it from mysqld
+	if !agent.gotMysqlPort {
```

Comment:
optional: I still think this would be much more readable if it was in its own method where you can return early in case of an error.